embedded-graphics/src/primitives/sector.rs

```diff
@@ -0,0 +1,797 @@
+    ///
+    /// ```
+    /// # use embedded_graphics::primitives::Sector;
+    /// # use embedded_graphics::prelude::*;
+    /// let sector = Sector::new(Point::new(5, 10), 10, 0.0.deg(), 90.0.deg());
+    /// let moved = sector.translate(Point::new(10, 10));
+    ///
+    /// assert_eq!(moved.top_left, Point::new(15, 20));
+    /// ```
+    fn translate(&self, by: Point) -> Self {
+        Self {
+            top_left: self.top_left + by,
+            ..*self
+        }
+    }
+
+    /// Translate the sector from its current position to a new position by (x, y) pixels.
+    ///
+    /// ```
+    /// # use embedded_graphics::primitives::Sector;
+    /// # use embedded_graphics::prelude::*;
+    /// let mut sector = Sector::new(Point::new(5, 10), 10, 0.0.deg(), 90.0.deg());
+    /// sector.translate_mut(Point::new(10, 10));
+    ///
+    /// assert_eq!(sector.top_left, Point::new(15, 20));
+    /// ```
+    fn translate_mut(&mut self, by: Point) -> &mut Self {
+        self.top_left += by;
+
+        self
+    }
+}
+
+/// Iterator over all points inside the sector.
+#[derive(Copy, Clone, PartialEq, PartialOrd, Debug)]
+pub struct Points {
+    iter: DistanceIterator,
+    threshold: u32,
+}
+
+impl Points {
+    fn new(sector: &Sector) -> Self {
+        let threshold = circle::diameter_to_threshold(sector.diameter);
+
+        Self {
+            iter: DistanceIterator::new(&sector),
+            threshold,
+        }
+    }
+}
+
+impl Iterator for Points {
+    type Item = Point;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let threshold = self.threshold;
+        self.iter
+            .find(|(_, distance)| *distance < threshold)
+            .map(|(point, _)| point)
+    }
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+enum IterState {
+    ARC,
+    LINES,
+    DONE,
+}
+
+/// Pixel iterator for each pixel in the sector border
+#[derive(Copy, Clone, PartialEq, PartialOrd, Debug)]
+pub struct StyledSectorIterator<C>
+where
+    C: PixelColor,
+{
+    iter: DistanceIterator,
+
+    outer_threshold: u32,
+    outer_color: Option<C>,
+
+    inner_threshold: u32,
+    inner_color: Option<C>,
+
+    line_a_iter: ThickLineIterator,
+    line_b_iter: ThickLineIterator,
+
+    state: IterState,
+}
+
+impl<C> StyledSectorIterator<C>
+where
+    C: PixelColor,
+{
+    fn new(styled: &Styled<Sector, PrimitiveStyle<C>>) -> Self {
+        let Styled { primitive, style } = styled;
+
+        let stroke_area = primitive.expand(style.outside_stroke_width());
+        let fill_area = primitive.shrink(style.inside_stroke_width());
+
+        let inner_threshold = circle::diameter_to_threshold(fill_area.diameter);
+        let outer_threshold = circle::diameter_to_threshold(stroke_area.diameter);
+
+        let line_a = stroke_area.line_from_angle(styled.primitive.angle_start);
+        let line_b = stroke_area.line_from_angle(styled.primitive.angle_end());
```

Comment:
Won't this draw pixels near the center of the arc multiple times? I'm ok to leave it as is for this PR, but we should maybe open an issue to optimise arc filling. Also, other primitives AFAIK make an effort to not draw any two pixels twice so it would be good to have `Sector` mirror this behaviour.